web_poet/page_inputs/stats.py

```diff
@@ -0,0 +1,40 @@
+from typing import Any, Dict, Protocol
+
+
+class StatCollectorProtocol(Protocol):
+    def set(self, key: str, value: Any) -> None:  # noqa: D102
+        pass  # pragma: no cover
+
+    def inc(self, key: str, value: int = 1) -> None:  # noqa: D102
+        pass  # pragma: no cover
+
+
+class DummyStatCollector:
+    def __init__(self):
+        self._stats: Dict[str, Any] = {}
+
+    def set(self, key: str, value: Any) -> None:  # noqa: D102
+        self._stats[key] = value
+
+    def inc(self, key: str, value: int = 1) -> None:  # noqa: D102
+        if key in self._stats:
+            assert isinstance(self._stats[key], int)
+            self._stats[key] += value
+        else:
+            self._stats[key] = value
+
+
+class Stats:
+    """Stat collector."""
```

Comment:
It'd be nice to document it a bit more, at least the the stat_collector approach. It seems "frameworks" section is the right place for it.